cmd/tasks.go

```diff
@@ -4,34 +4,51 @@ import (
 	"crypto"
 	"fmt"
 
-	"github.com/anchore/syft/syft/artifact"
-
-	"github.com/anchore/syft/syft/sbom"
-
+	"github.com/anchore/syft/internal/config"
 	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/artifact"
 	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/sbom"
 	"github.com/anchore/syft/syft/source"
 )
 
-type powerUserTask func(*sbom.Artifacts, *source.Source) ([]artifact.Relationship, error)
+type task func(*sbom.Artifacts, *source.Source) ([]artifact.Relationship, error)
 
-func powerUserTasks() ([]powerUserTask, error) {
-	var tasks []powerUserTask
+func tasks(appConfig *config.Application) ([]task, error) {
+	var tasks []task
 
-	generators := []func() (powerUserTask, error){
+	generators := []func() (task, error){
 		catalogPackagesTask,
-		catalogFileMetadataTask,
-		catalogFileDigestsTask,
-		catalogSecretsTask,
-		catalogFileClassificationsTask,
-		catalogContentsTask,
+		// catalogFileMetadataTask,
+		// catalogFileDigestsTask,
+		// catalogSecretsTask,
+		// catalogFileClassificationsTask,
+		// catalogContentsTask,
+	}
+
+	if appConfig.FileClassification.Cataloger.Enabled {
```

Comment:
currently the tasks themselves are responsible for including or excluding tasking based on configuration:  see https://github.com/anchore/syft/pull/636/files#diff-1b5d75686c7c5f1907c7542be728036ec235e8082046ba12206d38e63945402aL64-L66